financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsConsumerSessionRepository.kt

```diff
@@ -0,0 +1,108 @@
+package com.stripe.android.financialconnections.repository
+
+import com.stripe.android.core.Logger
+import com.stripe.android.core.networking.ApiRequest
+import com.stripe.android.model.ConsumerSession
+import com.stripe.android.model.ConsumerSessionLookup
+import com.stripe.android.repository.ConsumersApiService
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import java.util.Locale
+
+internal interface FinancialConnectionsConsumerSessionRepository {
+
+    suspend fun getCachedConsumerSession(): ConsumerSession?
+    suspend fun lookupConsumerSession(email: String?): ConsumerSessionLookup
+    suspend fun startConsumerVerification(consumerSessionClientSecret: String): ConsumerSession
+    suspend fun confirmConsumerVerification(
+        consumerSessionClientSecret: String,
+        verificationCode: String,
+    ): ConsumerSession
+
+    companion object {
+        operator fun invoke(
+            consumersApiService: ConsumersApiService,
+            apiOptions: ApiRequest.Options,
+            locale: Locale?,
+            logger: Logger,
+        ): FinancialConnectionsConsumerSessionRepository =
+            FinancialConnectionsConsumerSessionRepositoryImpl(
+                consumersApiService,
+                apiOptions,
+                locale,
+                logger,
+            )
+    }
+}
+
+private class FinancialConnectionsConsumerSessionRepositoryImpl(
+    private val consumersApiService: ConsumersApiService,
+    private val apiOptions: ApiRequest.Options,
+    private val locale: Locale?,
+    private val logger: Logger,
+) : FinancialConnectionsConsumerSessionRepository {
+
+    private val mutex = Mutex()
+    private var cachedConsumerSession: ConsumerSession? = null
```

Comment:
consumer session will be used and updated across panes, used the same approach (in-memory cache repository) than in the [manifest repository](https://github.com/stripe/stripe-android/blob/3c976825/financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsManifestRepository.kt#L1)